app/src/main/java/com/framgia/music_51/screen/detail_genre/DetailGenreActivity.java

```diff
@@ -2,20 +2,78 @@
 
 import android.content.Context;
 import android.content.Intent;
+import android.databinding.DataBindingUtil;
+import android.os.Parcelable;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.util.Log;
+import android.view.View;
 
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.request.RequestOptions;
 import com.framgia.music_51.R;
+import com.framgia.music_51.data.model.MusicResponse;
+import com.framgia.music_51.data.model.Track;
+import com.framgia.music_51.databinding.ActivityDetalGenreBinding;
+import com.framgia.music_51.screen.BindingUtils;
 
-public class DetailGenreActivity extends AppCompatActivity {
+import java.util.ArrayList;
+import java.util.List;
 
-    public static Intent getIntent(Context context) {
-        return new Intent(context, DetailGenreActivity.class);
+public class DetailGenreActivity extends AppCompatActivity implements View.OnClickListener {
+    private static final String EXTRA_LIST_TRACK = "EXTRA_LIST_TRACK";
+    private static final String EXTRA_NAME_GENRE = "EXTRA_NAME_GENRE";
+    private static final String EXTRA_IMAGE = "EXTRA_IMAGE";
+    private static final String BUNDLE_GENRE = "BUNDLE_GENRE";
+    private static final int INDEX = 0;
+    private DetailAdapter mAdapter;
+    private List<Track> mTracks;
+    private String mTitle;
+    private ActivityDetalGenreBinding mBinding;
+    private String mUrl;
+
+    public static Intent getIntent(Context context, MusicResponse genres) {
+        Bundle bundle = new Bundle();
+        bundle.putParcelableArrayList(EXTRA_LIST_TRACK, (ArrayList<? extends Parcelable>) genres.getTracks(genres));
+        bundle.putString(EXTRA_NAME_GENRE, genres.getGenre());
+        bundle.putString(EXTRA_IMAGE, genres.getCollections().get(INDEX).getTrack().getArtworkUrl());
+        Intent intent = new Intent(context, DetailGenreActivity.class);
+        intent.putExtra(BUNDLE_GENRE, bundle);
+        return intent;
     }
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_detal_genre);
+        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_detal_genre);
+        mAdapter = new DetailAdapter();
+        showCollection(getMusicGenreIntent());
+        mBinding.recyclerDetailGenre.setAdapter(mAdapter);
+        mBinding.buttonBack.setOnClickListener(this);
+        mBinding.recyclerDetailGenre.setLayoutManager(new LinearLayoutManager(this));
+    }
+
+    private List<Track> getMusicGenreIntent() {
+        Bundle bundle = getIntent().getBundleExtra(BUNDLE_GENRE);
+        mTracks = bundle.getParcelableArrayList(EXTRA_LIST_TRACK);
+        mTitle = bundle.getString(EXTRA_NAME_GENRE);
+        mUrl = bundle.getString(EXTRA_IMAGE);
+        return mTracks;
```

Comment:
method này sao lại return mTracks?
e nên viết như sau
```suggestion
        private List<Track> getMusicGenreIntent() {
        Bundle bundle = getIntent().getBundleExtra(BUNDLE_GENRE);
       List<Track> tracks = bundle.getParcelableArrayList(EXTRA_LIST_TRACK);
        mTitle = bundle.getString(EXTRA_NAME_GENRE);
        mUrl = bundle.getString(EXTRA_IMAGE);
        return tracks;
```